app/javascript/components/student/Editor.tsx

```diff
@@ -207,12 +209,20 @@ export function Editor({
         body: JSON.stringify({ files: files }),
       })
         .then((json: any) => {
+          if (!isMountedRef.current) {
```

Comment:
If we have this pattern at multiple places (where there is a `isMountedRef ` and `AbortController`), we could consider encoding this in some helper function at some point.

Reply:
I agree! I created https://github.com/exercism/v3-website/issues/255 for this one.

Reply:
@SleeplessByte Have you got this as a custom hook?